drivers/flash/soc_flash_nrf.c

```diff
@@ -95,6 +99,24 @@ static inline bool is_addr_valid(off_t addr, size_t len)
 	return true;
 }
 
+#ifdef CONFIG_SOC_FLASH_NRF_UICR
+static inline bool is_uicr_addr_valid(off_t addr, size_t len)
+{
+	if (addr + len > (off_t)NRF_UICR + sizeof(*NRF_UICR) ||
```

Comment:
it seems that you can get function to return true if you use invalid `addr` and `len` that just overflow the right way

Reply:
Just to be clear - x is for this part

Reply:
Thanks a lot for spot that! I see now that the regular access check is affected as well.